### PR TITLE
Data loading nonfloat bug

### PR DESCRIPTION
**Pull request by JGarciaCondado**
Added changes to ensure that users know better when there files are not properly formatted. Which includes:
- Changing clinical files to be 0 or 1
- Only allowing floats
- Files are now checked to ensure no strings are found and also in the case of clinical file to ensure no rows with all 0 or 1.

### CODE REVIEW

File: src/ageml/ui.py [hunk 1]
```diff
@@ -314,6 +314,14 @@ def load_features(self, required=False):
         if "age" not in df:
             raise KeyError("Features file must contain a column name 'age', or any other case-insensitive variation.")
 
+        # Check that columns are dtypes float or int
+        error_cols = []
+        for col in df.columns:
+            if df[col].dtype not in [float, int]:
+                error_cols.append(col)
+        if error_cols != []:
+            raise TypeError("Columns must be float or int type: %s" % (error_cols))
```
Comment (itellaetxe) [inline src/ageml/ui.py:323]: Include which file we are referring to -> features

File: src/ageml/ui.py [hunk 2]
```diff
@@ -342,13 +350,21 @@ def load_covariates(self, required=False):
         # Set covariate flag
         self.flags['covariates'] = True
 
+        # Check that columns are dtypes float or int
+        error_cols = []
+        for col in df.columns:
+            if df[col].dtype not in [float, int]:
+                error_cols.append(col)
+        if error_cols != []:
+            raise TypeError("Columns must be float or int type: %s" % (error_cols))
```
Comment (itellaetxe) [inline src/ageml/ui.py:359]: Include which file we are referring to -> covariates

File: src/ageml/ui.py [hunk 3]
```diff
@@ -374,17 +390,31 @@ def load_clinical(self, required=False):
...
+            # Check if all values in the column are either 0 or 1
+            if df[column].isin([0, 1]).all():
+                # Convert the column to boolean type
+                df[column] = df[column].astype(bool)
+            else:
+                error_cols.append(column)
+
+        # Raise an error if the column contains values other than 0 or 1
+        if error_cols != []:
+            raise TypeError(f"Columns: {error_cols} contains values other than 0 and 1.")
```
Comment (itellaetxe) [inline src/ageml/ui.py:406]: Include which file we are referring to -> clinical

File: src/ageml/ui.py [hunk 4]
```diff
@@ -374,17 +390,31 @@ def load_clinical(self, required=False):
...
         
         # Check that all columns have at least two subjects and show which column
         for col in df.columns:
-            if df[col].sum() == 0:
-                raise ValueError("Clinical column %s has no subjects." % col)
+            if df[col].sum() < 2:
+                raise ValueError("Clinical column %s has less than two subjects." % col)
```
Comment (itellaetxe) [inline src/ageml/ui.py:411]: Perhaps we could check that for every clinical category we have at least one subject. Makes this more robust and prone to less errors upstream

File: src/ageml/ui.py [hunk 5]
```diff
@@ -411,6 +441,14 @@ def load_factors(self, required=False):
         elif df is None:
             return df
         
+        # Check that columns are dtypes float or int
+        error_cols = []
+        for col in df.columns:
+            if df[col].dtype not in [float, int]:
+                error_cols.append(col)
+        if error_cols != []:
+            raise TypeError("Columns must be float or int type: %s" % (error_cols))
```
Comment (itellaetxe) [inline src/ageml/ui.py:450]: Include which file we are referring to -> factors

File: src/ageml/ui.py [hunk 6]
```diff
@@ -432,6 +470,14 @@ def load_ages(self, required=False):
...
         elif df is None:
             return df
+        
+        # Check that columns are dtypes float or int
+        error_cols = []
+        for col in df.columns:
+            if df[col].dtype not in [float, int]:
+                error_cols.append(col)
+        if error_cols != []:
+            raise TypeError("Columns must be float or int type: %s" % (error_cols))
```
Comment (itellaetxe) [inline src/ageml/ui.py:480]: Include which file we are referring to -> ages

File: tests/test_ageml/test_ui.py [hunk 7]
```diff
@@ -261,6 +261,66 @@ def test_load_data_age_not_column(dummy_interface, features):
     assert exc_info.value.args[0] == error_message
 
 
+def test_load_data_not_float(dummy_interface, features):
+    # Change item to string
+    features.loc[2, "feature1"] = "asdf"
```
Comment (itellaetxe) [inline tests/test_ageml/test_ui.py:266]: unfunny placeholder :(

File: tests/test_ageml/test_ui.py [hunk 8]
```diff
@@ -261,6 +261,66 @@ def test_load_data_age_not_column(dummy_interface, features):
...
+    features.loc[3, "feature2"] = "asdf"
+    features_path = create_csv(features, dummy_interface.dir_path)
+    dummy_interface.args.features = features_path
+
+    # Test error risen
+    with pytest.raises(TypeError) as exc_info:
+        dummy_interface.load_data()
+    assert exc_info.type == TypeError
+    error_message = "Columns must be float or int type: ['feature1', 'feature2']"
```
Comment (itellaetxe) [inline tests/test_ageml/test_ui.py:275]: Match with suggested error message (should contain "features")

File: tests/test_ageml/test_ui.py [hunk 9]
```diff
@@ -261,6 +261,66 @@ def test_load_data_age_not_column(dummy_interface, features):
...
+    factors.loc[2, "factor1"] = "asdf"
+    factors.loc[3, "factor2"] = "asdf"
+    factors_path = create_csv(factors, dummy_interface.dir_path)
+    dummy_interface.args.factors = factors_path
+
+    # Test error risen
+    with pytest.raises(TypeError) as exc_info:
+        dummy_interface.load_data()
+    assert exc_info.type == TypeError
+    error_message = "Columns must be float or int type: ['factor1', 'factor2']"
```
Comment (itellaetxe) [inline tests/test_ageml/test_ui.py:290]: Match with suggested error message (should contain "factors")

File: tests/test_ageml/test_ui.py [hunk 10]
```diff
@@ -261,6 +261,66 @@ def test_load_data_age_not_column(dummy_interface, features):
...
+    # Change item to string
+    covariates.loc[2, 'sex'] = 'asdf'
+    covariates_path = create_csv(covariates, dummy_interface.dir_path)
+    dummy_interface.args.covariates = covariates_path
+
+    # Test error risen
+    with pytest.raises(TypeError) as exc_info:
+        dummy_interface.load_data()
+    assert exc_info.type == TypeError
+    error_message = "Columns must be float or int type: ['sex']"
```
Comment (itellaetxe) [inline tests/test_ageml/test_ui.py:304]: Match with suggested error message (should contain "covariates")

File: tests/test_ageml/test_ui.py [hunk 11]
```diff
@@ -261,6 +261,66 @@ def test_load_data_age_not_column(dummy_interface, features):
...
+    ages.loc[3, "corrected_age_all"] = "asdf"
+    ages.loc[4, "delta_all"] = "asdf"
+    ages_path = create_csv(ages, dummy_interface.dir_path)
+    dummy_interface.args.ages = ages_path
+
+    # Test error risen
+    with pytest.raises(TypeError) as exc_info:
+        dummy_interface.load_data()
+    assert exc_info.type == TypeError
+    error_message = "Columns must be float or int type: ['predicted_age_all', 'corrected_age_all', 'delta_all']"
```
Comment (itellaetxe) [inline tests/test_ageml/test_ui.py:320]: Match with suggested error message (should contain "ages")

File: tests/test_ageml/test_ui.py [hunk 12]
```diff
@@ -325,34 +385,34 @@ def test_load_data_clinical_not_boolean(dummy_interface, clinical):
     with pytest.raises(TypeError) as exc_info:
         dummy_interface.load_data()
     assert exc_info.type == TypeError
-    assert exc_info.value.args[0] == "Clinical columns must be boolean type. Check that all values are encoded as 'True' or 'False'."
+    assert exc_info.value.args[0] == "Columns: ['cn', 'group1'] contains values other than 0 and 1."
```
Comment (itellaetxe) [inline tests/test_ageml/test_ui.py:388]: Match with suggested error message (should contain "clinical")

**Review by itellaetxe — CHANGES_REQUESTED** [review]

File: src/ageml/ui.py (hunk 4 above)
Comment (JGarciaCondado) [reply src/ageml/ui.py:411]: Indeed i am doing that

**Review by itellaetxe — APPROVED** [review]
LGTM, approving

Merging

**Pull request merged**